### PR TITLE
Upload to annotator from disk

### PR DESCRIPTION
**Pull request by andrei-msd**

### CODE REVIEW

File: frontend-react/src/components/annotate.js [hunk 1]
```diff
@@ -29,15 +28,20 @@ class Annotate extends Component {
...
+                    <input type="file" id="file" multiple={true} accept="image/*" className="fileInputButton" onChange={this.fileSelectedHandler} />
+                    <label htmlFor="file" className="fileLabel">
+                      Add Images
+                    </label>
+                  </form>
                 <div className="imageDisplay">
                   <ReactImageAnnotate
-                    taskDescription="# Draw region around each Kiwifruit bunch"
-                    images= {data} 
-                    regionClsList={["kiwi fruit", "leaf"]}
```
Comment (yaserazfar) [inline frontend-react/src/components/annotate.js]: I am no longer able to add a label to annotations. I think we may need to leave some default annotations or the empty string `regionClsList={[""]}` inside of here.

File: frontend-react/src/components/annotate.js [hunk 2]
```diff
@@ -2,70 +2,72 @@ import React, { Component } from 'react';
...
-                        Select Image
+                      Add Images
                     </label>
-                </form>
+                  </form>
                 <div className="imageDisplay">
+                  <ReactImageAnnotate
+                    images= {data}
+                    key = {data}
+                    regionClsList={[]}
```
Comment (yaserazfar) [inline frontend-react/src/components/annotate.js]: I am no longer able to add a label to annotations. I think we may need to leave some default annotations or the empty string `regionClsList={[""]}`

**Review by yaserazfar — DISMISSED** [review]
I think we should also add a TODO comment somewhere in annotate.js to keep us aware of the issue where uploading a new image currently will reset annotations. This probably cannot be fixed until we have made some progress with the JS Client.

Also we can probably make the "upload images" button bigger to improve the UI.

Apart from these minor points, it looks great.

**Review by yaserazfar — APPROVED** [review]
LGTM.

We need to make adding annotations strings possible without leaving those default values in `regionCLSList` but that can be worked on later and it works for the changes we need right now.

File: frontend-react/src/components/annotate.js [hunk 3]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
 
-          let reader = new FileReader();
-          let file = e.target.files[0];
+  fileSelectedHandler = event => {
+    var images = [];
+    let files = event.target.files;
+    var img = new Image();
+    for (let i = 0; i < files.length; i++) {
+        img.src = URL.createObjectURL(files[i]);
+        images.push({src: img.src, name: files[i].name});
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
        images.push({src: img.src, name: files[i].name});
```

File: frontend-react/src/components/annotate.js [hunk 4]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
+    for (let i = 0; i < files.length; i++) {
+        img.src = URL.createObjectURL(files[i]);
+        images.push({src: img.src, name: files[i].name});  
+    }
 
-          reader.onloadend = () => {
-              this.setState({
-                  file: file,
-                  imagePreviewUrl: reader.result
-              });
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js:32]: I think this style looks better...

File: frontend-react/src/components/annotate.js [hunk 5]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
-          };
+    this.setState({data: this.state.data.concat(images)});
+    this.setState({uploaded: true});
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]: ... than this

File: frontend-react/src/components/annotate.js [hunk 6]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
+  }
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
  }
```

File: frontend-react/src/components/annotate.js [hunk 7]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
-            $imagePreview = (<img src={imagePreviewUrl} />);
-        }
-
-        return (
-            <div className="border">
-                <form className="selectForm" onSubmit={this._handleSubmit}>
-                    <input type="file" id="file" className="fileInputButton" onChange={this._handleImageChange} />
+      render() {
+        let { data } = this.state;
+        let { uploaded } = this.state;
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]: Same as above.

File: frontend-react/src/components/annotate.js [hunk 8]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
 
-                    {/* <ReactImageAnnotate
-                        selectedImage="https://i.imgur.com/SY4z4Cx.jpg"
-                        taskDescription="# Draw region around each Kiwifruit bunch"
-                        images={[{ src: "https://i.imgur.com/SY4z4Cx.jpg", name: "Kiwifruit.png" }]}
-                        regionClsList={["Kiwifruit", "Human Face"]}
-                        enabledTools="create-polygon, create-box"
-                    /> */}
+                </div>
+
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
```

File: frontend-react/src/components/annotate.js [hunk 9]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
-                        enabledTools="create-polygon, create-box"
-                    /> */}
+                </div>
+                
+              </div>
+          );}
+        else{
+            return (
+                <div className="border">
+
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
```

File: frontend-react/src/components/annotate.js [hunk 10]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
+                </div>
+                
+              </div>
+          );}
+        else{
+            return (
+                <div className="border">
+         
+                  <div className="imageDisplay">
+
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
```

File: frontend-react/src/components/annotate.js [hunk 11]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
+                    <form className="selectForm" id="formUpload">
+                        <input type="file" id="file" multiple={true} accept="image/*" className="fileInputButton" onChange={this.fileSelectedHandler} />
+                        <label htmlFor="file" className="fileLabel">
+                            Upload Images
+                        </label>
+                    </form>
 
-                    {$imagePreview}
+                  </div>
+
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
```

File: frontend-react/src/components/annotate.js [hunk 12]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
-                        taskDescription="# Draw region around each Kiwifruit bunch"
-                        images={[{ src: "https://i.imgur.com/SY4z4Cx.jpg", name: "Kiwifruit.png" }]}
-                        regionClsList={["Kiwifruit", "Human Face"]}
-                        enabledTools="create-polygon, create-box"
-                    /> */}
+                </div>
+                
+              </div>
+          );}
+        else{
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
        else {
```

File: frontend-react/src/components/annotate.js [hunk 13]
```diff
@@ -2,70 +2,74 @@ import React, { Component } from 'react';
...
+                    </form>
 
-                    {$imagePreview}
+                  </div>
+                
                 </div>
-            </div>
-        );
-    }
+            )
```
Comment (michaeliwaikato) [inline frontend-react/src/components/annotate.js]:
```suggestion
            );
```

File: frontend-react/src/style/annotate.css [hunk 14]
```diff
@@ -20,9 +20,12 @@
   background-image: linear-gradient(#009DFF, #59BFFF);
   color: #fff;
   padding:10px;
+  padding-right: 40px;
+  padding-left: 40px;
   border-radius: 5px;
-  font-size:15px;
+  font-size:20px;
   font-weight: bold;
+
```
Comment (michaeliwaikato) [inline frontend-react/src/style/annotate.css]:
```suggestion
```

File: frontend-react/src/style/annotate.css [hunk 15]
```diff
@@ -50,6 +54,9 @@ img {
...
 
 .selectForm {
-  padding-top: 25px;
-  padding-bottom: 20px;
+  margin-top: 20px;
 }
+
+#formUpload {
+  margin-top: 20%;
+}
```
Comment (michaeliwaikato) [inline frontend-react/src/style/annotate.css]: We should have newlines at the end of files.

**Review by michaeliwaikato — CHANGES_REQUESTED** [review]

**Review by michaeliwaikato — APPROVED** [review]

**Pull request merged**